aws-java-sdk-core/src/main/java/com/amazonaws/internal/SdkThreadLocalsRegistry.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class SdkThreadLocalsRegistry {
+    
+    private static final List<ThreadLocal<?>> threadLocals = new ArrayList<ThreadLocal<?>>();
```

Comment:
might be nice to use Map(className to ThreadLocal maybe?) here so that customers can remove ThreadLocal use for certain class only.

Reply:
I only have a use case (application shutdown/restart with a container managed thread pool) for removal of all. Can you suggest a use case for that scenario?